### PR TITLE
zigbee: samples: Add application template sample

### PR DESCRIPTION
**Pull request by tomchy**
Introduce a bare minimum sample that can be used as a starting point for custom Zigbee applications.

Signed-off-by: Tomasz Chyrowicz <tomasz.chyrowicz@nordicsemi.no>

### CODE REVIEW

File: samples/zigbee/template/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.13.1)
```
Comment (rlubos) [inline samples/zigbee/template/CMakeLists.txt:7]: `3.20.0` as per https://github.com/nrfconnect/sdk-nrf/pull/5262

File: samples/zigbee/template/prj.conf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+# Enable nRF ECB driver
+CONFIG_CRYPTO=y
+CONFIG_CRYPTO_NRF_ECB=y
+CONFIG_CRYPTO_INIT_PRIORITY=80
+
+# Cryptocell is not supported through CSPRNG driver API: NCSDK-4813
+CONFIG_ENTROPY_CC3XX=n
+
+#Networking
```
Comment (rlubos) [inline samples/zigbee/template/prj.conf:39]: nit:
```suggestion
# Networking
```

File: samples/zigbee/template/prj_nrf5340dk_nrf5340_cpuapp.conf [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#
```
Comment (rlubos) [inline samples/zigbee/template/prj_nrf5340dk_nrf5340_cpuapp.conf:1]: I just wonder, this is pretty similar to the default `prj.conf` - did you consider creating `boards/nrf5340dk_nrf5340_cpuapp.conf` and include only differences in there? Such a config file should be applied automatically for that specifc board.

File: samples/zigbee/template/src/main.c [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+/* Button used to enter the Identify mode. */
+#define IDENTIFY_MODE_BUTTON            DK_BTN4_MSK
+
+
+LOG_MODULE_REGISTER(app);
+
+/* Main application customizable context.
+ * Stores all settings and static values.
+ */
+struct zb_device_ctx_s {
```
Comment (rlubos) [inline samples/zigbee/template/src/main.c:45]: nit: We don't typically add `_s` suffix in NCS codebase (I guess it still falls under "hungarian notation")?

File: samples/zigbee/template/src/main.c [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+	ZB_AF_SET_IDENTIFY_NOTIFICATION_HANDLER(APP_TEMPLATE_ENDPOINT, identify_cb);
+
+	/* Start Zigbee default thread */
+	zigbee_enable();
+
+	LOG_INF("Zigbee application template started");
+
+	while (true) {
+		/* Spin forever */
+		k_sleep(K_FOREVER);
```
Comment (rlubos) [inline samples/zigbee/template/src/main.c:236]: You can do this, but I just wonder - why not simply exit the thread? It shall not impact other thread's operation.

**Review by rlubos — COMMENTED** [review]
Looks good, just some nitpicking

**Review by greg-fer — APPROVED** [review]
Why are we going for an implementation that includes FEM? Isn't this against the idea of a minimal implementation

**Comment by greg-fer** [comment]
Please remember to add an entry to changelog documenting this addition.

File: samples/zigbee/template/src/main.c (hunk 4 above)
Comment (tomchy) [reply samples/zigbee/template/src/main.c:45]: Removed :slightly_smiling_face:

File: samples/zigbee/template/prj_nrf5340dk_nrf5340_cpuapp.conf (hunk 3 above)
Comment (tomchy) [reply samples/zigbee/template/prj_nrf5340dk_nrf5340_cpuapp.conf:1]: I'll try that. Thanks for the suggestion!

File: samples/zigbee/template/src/main.c [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+	if (err) {
+		LOG_ERR("Cannot init buttons (err: %d)", err);
+	}
+
+	err = dk_leds_init();
+	if (err) {
+		LOG_ERR("Cannot init LEDs (err: %d)", err);
+	}
+}
+
```
Comment (sebastiandraus) [inline samples/zigbee/template/src/main.c:179]: Nit: double newline

**Review by sebastiandraus — APPROVED** [review]

**Comment by tomchy** [comment]
> Please remember to add an entry to changelog documenting this addition.

@greg-fer Should I add it in this PR or wait for the dedicated PR close to the next release date?

**Review by rlubos — APPROVED** [review]

File: samples/zigbee/template/CMakeLists.txt (hunk 1 above)
Comment (tomchy) [reply samples/zigbee/template/CMakeLists.txt]: Currently it fails the build:
```
CMake Error at CMakeLists.txt:7 (cmake_minimum_required):
  CMake 3.20.0 or higher is required.  You are running version 3.17.3
```

Comment (rlubos) [reply samples/zigbee/template/CMakeLists.txt]: @tejlmand Is this being covered with regards to #5262? It seems to be an issue with our build servers running older CMake.

Comment (tomchy) [reply samples/zigbee/template/CMakeLists.txt]: I'll give it the last chance. If it fails - I am going to revert it to the 3.17.3. It is counterproductive to wait for this PR (we are waiting for it for 7 days now and the CI is still red there).

Comment (tomchy) [reply samples/zigbee/template/CMakeLists.txt]: Reverted.

**Comment by tomchy** [comment]
@greg-fer Changelog updated.

**Pull request merged**